core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java

```diff
@@ -86,32 +95,68 @@ public static void addSourceProvider(SmallRyeConfigBuilder builder, ConfigSource
         }
     }
 
-    static final class EnvConfigSource implements ConfigSource {
+    static class EnvConfigSource implements ConfigSource {
         static final Pattern REP_PATTERN = Pattern.compile("[^a-zA-Z0-9_]");
 
-        private final int ordinal;
-
-        EnvConfigSource(final int ordinal) {
-            this.ordinal = ordinal;
+        EnvConfigSource() {
         }
 
         public int getOrdinal() {
-            return ordinal;
+            return 300;
         }
 
         public Map<String, String> getProperties() {
             return Collections.emptyMap();
         }
 
         public String getValue(final String propertyName) {
-            return System.getenv(REP_PATTERN.matcher(propertyName.toUpperCase(Locale.ROOT)).replaceAll("_"));
+            return getRawValue(REP_PATTERN.matcher(propertyName.toUpperCase(Locale.ROOT)).replaceAll("_"));
+        }
+
+        String getRawValue(final String name) {
+            return System.getenv(name);
         }
 
         public String getName() {
             return "System environment";
         }
     }
 
+    static class DotEnvConfigSource extends EnvConfigSource {
+        private final Map<String, String> values;
+
+        DotEnvConfigSource() {
+            this(Paths.get(System.getProperty("user.dir", "."), ".env"));
+        }
+
+        DotEnvConfigSource(Path path) {
+            Map<String, String> values = new HashMap<>();
+            try (InputStream is = Files.newInputStream(path)) {
+                try (InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8)) {
+                    final Properties properties = new Properties();
+                    properties.load(isr);
+                    for (String name : properties.stringPropertyNames()) {
+                        values.put(name, properties.getProperty(name));
+                    }
+                }
+            } catch (IOException ignored) {
```

Comment:
Not a big deal, but maybe it would be nice to debug log any IOExceptions thrown here

Reply:
Fixed :+1: